contracts/okx/AnteOKXEthReservesTest.sol

```diff
@@ -0,0 +1,362 @@
+    event AnteReservesAdded(address indexed token, address wallet);
+
+    /// @notice Emitted when test owner commits a Chainlink price feed update
+    /// @param token the address of the token asset the feed is for
+    /// @param oldFeed the address of the previous feed that is being replaced
+    /// @param newFeed the new address of the price feed
+    event AntePriceFeedPendingUpdate(address indexed token, address oldFeed, address newFeed);
+
+    /// @notice Emitted when test owner updates a Chainlink price feed
+    /// @param token the address of the token asset the feed is for
+    /// @param oldFeed the address of the previous feed that is being replaced
+    /// @param newFeed the new address of the price feed
+    event AntePriceFeedUpdated(address indexed token, address oldFeed, address newFeed);
+
+    /// @notice Emitted when test owner commits a failure threshold update
+    /// @param oldThreshold the previous failure threshold value (USD)
+    /// @param newThreshold the new failure threshold value (USD)
+    event AnteTestPendingUpdate(uint256 oldThreshold, uint256 newThreshold);
+
+    /// @notice Emitted when test owner updates the failure threshold value
+    /// @param oldThreshold the previous failure threshold value (USD)
+    /// @param newThreshold the new failure threshold value (USD)
+    event AnteTestFailureUpdated(uint256 oldThreshold, uint256 newThreshold);
+
+    // list of tokens to check
+    address[] public tokens;
+    // map tokens to price oracle
+    mapping(address => address) public priceFeeds;
+    // map tokens to wallets
+    mapping(address => address[]) public wallets;
+    // keep track of checked combinations and wallets
+    mapping(address => mapping(address => bool)) isChecked;
+    mapping(address => bool) isWalletTested;
+
+    // threshold asset balance for test to fail (will set in constructor)
+    uint256 public failureThreshold;
+
+    // Max wallets and tokens to check (to prevent unbounded gas usage)
+    uint256 public constant MAX_TOKENS_CHECKED = 10;
+    uint256 public constant MAX_WALLETS_PER_TOKEN = 50;
+
+    /// @notice minimum time between test updates by owner
+    uint256 public constant UPDATE_PRICEFEED_WAIT_PERIOD = 86400; // 1 day
+    uint256 public constant UPDATE_FAILURE_WAIT_PERIOD = 604800; // 7 days
+
+    // last timestamp test parameters were updated
+    uint256 public lastUpdated;
+    uint256 public updatePriceFeedCommitTime;
+    uint256 public updateThresholdCommitTime;
+
+    // variables to store updates to be enacted
+    address public newToken;
+    address public newPriceFeed;
+    uint256 public newThreshold;
+
+    constructor() {
+        protocolName = "OKX";
+
+        // Total Asset Value ~$3.9B on Ethereum according to
+        // https://portfolio.nansen.ai/dashboard/okx?chain=ETHEREUM
+        // as of 2022-12-23, $42M is ~1% of that and a nice number
+        failureThreshold = 69_000_000;
```

Comment:
i guess i thought it could be easier to grok "reserves under X amount" and the failure level is clear in that case but i guess for someone that doesn't have a clear sense of total reserves a % drop could be better